src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,34 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        UserData cachedData = cache.get(userId);
+        if (cachedData != null) {
+            return CompletableFuture.completedFuture(cachedData);
+        }
+
+        return CompletableFuture.supplyAsync(() -> {
+
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+            UserData userData = new UserData(userId, "Details for " + userId);
+            cache.put(userId, userData);
+            return userData;
+        }, executor);
```

Comment:
The use of `Thread.sleep(1000)` is generally discouraged within a CompletableFuture because it blocks the thread. If the intention is to simulate a delay or a long-running task, consider using a scheduled executor service to handle the delay non-blocking, or if you're simulating an I/O task, use proper asynchronous I/O operations.